File: _java-archive/acquisition-service/src/main/java/dev/kritchalach/acquisition/domain/AcquisitionStatus.java
package dev.kritchalach.acquisition.domain;

/**
 * Value Object: AcquisitionStatus
 * สถานะของการเข้าซื้อ property จาก seller โดยทีม Legal
 */
public enum AcquisitionStatus {
    SURVEYED,           // รับ survey data จาก Inventory แล้ว
    APPROVAL_REQUESTED, // ส่งขออนุมัติ CEO
    APPROVED,           // CEO อนุมัติให้ซื้อ
    CONTRACT_DRAFTED,   // ร่าง willing contract เสร็จ
    REJECTED            // CEO ไม่อนุมัติ / กระบวนการล้ม
}

File: acquisition-service/src/domain/AcquisitionStatus.ts
/**
 * Value Object: AcquisitionStatus
 * State machine ของการเข้าซื้อ property
 */
export const AcquisitionStatus = {
  SURVEYED: 'SURVEYED',
  APPROVAL_REQUESTED: 'APPROVAL_REQUESTED',
  APPROVED: 'APPROVED',
  CONTRACT_DRAFTED: 'CONTRACT_DRAFTED',
  REJECTED: 'REJECTED',
} as const;

export type AcquisitionStatus = (typeof AcquisitionStatus)[keyof typeof AcquisitionStatus];
